### PR TITLE
Clean up main.C and explicitly register dependent app

### PR DESCRIPTION
**Pull request by MengnanLi91**
Adapt to the new changes in MOOSE(https://github.com/idaholab/moose/pull/25994)

### CODE REVIEW

File: src/main.C [hunk 1]
```diff
@@ -13,33 +13,13 @@
...
 main(int argc, char * argv[])
 {
-  // Initialize MPI, solvers and MOOSE
-  MooseInit init(argc, argv);
-
-  // Register this application's MooseApp and any it depends on
-  BlackBearTestApp::registerApps();
-
-  // register inelastic strain
-  TensorMechanicsActionBase::addCartesianComponentOutput("inelastic_strain");
```
Comment (bwspenc) [inline src/main.C:36]: I just saw this got merged yesterday. 

How is `addCartesianComponentOutput()` call handled now?

Comment (bwspenc) [reply src/main.C:36]: ...or is it not necessary anymore? I don't actually know what the story is behind that -- this was the first time I had seen it.

Comment (MengnanLi91) [reply src/main.C:36]: My mistake, `addCartesianComponentOutput()` should stay. The MOOSE change doesn't handle `addCartesianComponentOutput()`

Comment (bwspenc) [reply src/main.C:36]: @MengnanLi91 Thanks! If you don't mind adding another PR that adds that back in, that would be great. I can merge it.

Comment (bwspenc) [reply src/main.C:36]: It would just go before the `Moose::main` call

Comment (MengnanLi91) [reply src/main.C:36]: Already in, check #382

**Pull request merged**